craft_parts/sequencer.py

```diff
@@ -81,3 +84,103 @@ def _add_step_actions(
     ) -> None:
         # TODO: implement this stub
         pass
+
+    def _process_dependencies(self, part: Part, step: Step) -> None:
+        prerequisite_step = steps.dependency_prerequisite_step(step)
+        if not prerequisite_step:
+            return
+
+        all_deps = parts.part_dependencies(part.name, part_list=self._part_list)
+
+        deps = {p for p in all_deps if self._sm.should_step_run(p, prerequisite_step)}
```

Comment:
Does order matter here?

Reply:
It shouldn't because if A depends on B and C, and B depends on C, B was already processed (parts are sorted by dependency in the Sequencer constructor) so C shouldn't run again. This is also the case in Snapcraft (corresponding code is in `snapcraft.internal.lifecycle._runner._Executor._prepare_step`).

Reply:
Thanks!